pandas/tests/test_frame.py

```diff
@@ -9140,6 +9140,13 @@ def test_replace_inplace(self):
         tsframe.replace([nan], [0], inplace=True)
         assert_frame_equal(tsframe, self.tsframe.fillna(0))
 
+        # GH 11698
+        # test for mixed data types.
+        df = pd.DataFrame([('-', pd.to_datetime('20150101')), ('a', pd.to_datetime('20150102'))])
+        df.replace('-', np.nan, inplace=True)
```

Comment:
use ``result = df.replace(....)``

never use ``inplace`` in a test, unless you are explicity testing something that involves it explicity. this is not pandonic at all.